addon/components/polaris-resource-list/bulk-actions.js

```diff
@@ -0,0 +1,535 @@
+    function() {
+      let {
+        promotedActions,
+        numberOfPromotedActionsToRender,
+      } = this.getProperties(
+        'promotedActions',
+        'numberOfPromotedActionsToRender'
+      );
+
+      if (
+        promotedActions &&
+        numberOfPromotedActionsToRender < promotedActions.length
+      ) {
+        return [...promotedActions].slice(numberOfPromotedActionsToRender);
+      }
+
+      return [];
+    }
+  ),
+
+  activatorLabel: computed(
+    'promotedActions',
+    'numberOfPromotedActionsToRender',
+    'measuring',
+    function() {
+      let {
+        promotedActions,
+        numberOfPromotedActionsToRender,
+        measuring,
+      } = this.getProperties(
+        'promotedActions',
+        'numberOfPromotedActionsToRender',
+        'measuring'
+      );
+
+      if (
+        !promotedActions ||
+        (promotedActions && numberOfPromotedActionsToRender === 0 && !measuring)
+      ) {
+        return 'Actions';
+      }
+
+      return 'More actions';
+    }
+  ),
+
+  shouldRenderActionsPopover: computed(
+    'actionSections',
+    'rolledInPromotedActions.length',
+    'measuring',
+    function() {
+      let {
+        actionSections,
+        rolledInPromotedActions,
+        measuring,
+      } = this.getProperties(
+        'actionSections',
+        'rolledInPromotedActions',
+        'measuring'
+      );
+
+      return Boolean(
+        actionSections || rolledInPromotedActions.length > 0 || measuring
+      );
+    }
+  ),
+
+  actionSections: computed('actionsCollection.[]', function() {
+    let actionsCollection = this.get('actionsCollection');
+
+    if (!actionsCollection || actionsCollection.length === 0) {
+      return null;
+    }
+
+    if (this.instanceOfBulkActionListSectionArray(actionsCollection)) {
+      return actionsCollection;
+    }
+
+    if (this.instanceOfBulkActionArray(actionsCollection)) {
+      return [{ items: actionsCollection }];
+    }
+  }),
+
+  combinedActions: computed(
+    'actionSections.[]',
+    'rolledInPromotedActions',
+    function() {
+      let combinedActions = [];
+      let { actionSections, rolledInPromotedActions } = this.getProperties(
+        'actionSections',
+        'rolledInPromotedActions'
+      );
+
+      let rolledInPromotedActionsHasLength = rolledInPromotedActions.length > 0;
+
+      if (actionSections && rolledInPromotedActionsHasLength) {
+        combinedActions = [
+          { items: rolledInPromotedActions },
+          ...actionSections,
+        ];
+      } else if (actionSections) {
+        combinedActions = actionSections;
+      } else if (rolledInPromotedActionsHasLength) {
+        combinedActions = [{ items: rolledInPromotedActions }];
+      }
+
+      return combinedActions;
+    }
+  ),
+
+  /**
+   * Callback when selectable state of list is changed
+   *
+   * @type {Function}
+   * @default noop
+   * @property onSelectModeToggle
+   */
+  onSelectModeToggle(/* selectMode **/) {},
```

Comment:
This and `onToggleAll` below should be tagged as `@public` and moved up beneath the other public props

Reply:
Ahh I left out the `@public` 🤦‍♂️ Was wondering why prettier was mad at me.